FormatIntWithPercent.java

```diff
@@ -0,0 +1,72 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FormatIntWithPercent {
+    private static final Logger LOG = LoggerFactory.getLogger(VMWApplianceConfigurationHelper.class);
+
+    public void runProcess(final ProcessBuilder pb, final String successOutput) {
+        int exitValue;
+        Process p = null;
+        final StringBuilder stdoutSB = new StringBuilder();
+        final StringBuilder stderrSB = new StringBuilder();
+        boolean isConfigured = false;
+        try {
+            p = pb.start();
+            try (BufferedReader stdoutBR = new BufferedReader(new InputStreamReader(p.getInputStream(),
+                    StandardCharsets.UTF_8));
+                 BufferedReader stderrBR = new BufferedReader(new InputStreamReader(p.getErrorStream(),
+                         StandardCharsets.UTF_8))) {
+                String line;
+                while ((line = stdoutBR.readLine()) != null) {
+                    stdoutSB.append(line);
+                    stdoutSB.append("\n");
+                    if (StringUtils.equalsIgnoreCase(line, successOutput)) {
+                        isConfigured = true;
+                    }
+                }
+
+                while ((line = stderrBR.readLine()) != null) {
+                    stderrSB.append(line);
+                    stderrSB.append("\n");
+                }
+
+                exitValue = p.waitFor();
+            }
+        } catch (final Exception e) {
+            throw new RuntimeException("Got exception running process to configure VMW appliance.", e);
+        } finally {
+            try {
+                if (p != null) {
+                    p.destroy();
+                    if (p.getInputStream() != null) {
+                        p.getInputStream().close();
+                    }
+                    if (p.getOutputStream() != null) {
+                        p.getOutputStream().close();
+                    }
+                    if (p.getErrorStream() != null) {
+                        p.getErrorStream().close();
+                    }
+                }
+            } catch (final IOException e) {
+                LOG.warn("Got exception in closing process streams", e);
+            }
+        }
+
+        if (exitValue != 0 || !isConfigured) {
+            final String msg = String.format(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: exitValue (declared at line 39). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 63. Consider formatting this data with &#39;%d&#39; instead.